Ginger/GingerCoreNET/GeneralLib/EmailReadGmailOperations.cs

```diff
@@ -61,15 +61,29 @@ public Task ReadEmails(EmailReadFilters filters, EmailReadConfig config, Action<
                 {
                     queryToImap = queryToImap.And(SearchQuery.DeliveredAfter(filters.ReceivedStartDate)).And(SearchQuery.DeliveredBefore(filters.ReceivedEndDate));
                 }
-                var query = new SearchQuery();
-
+                
                 // Add a condition to the search query.
                 IList<UniqueId> list = inbox.Search(queryToImap, cancellationToken);
 
                 foreach (var item in list)
                 {
                     MimeMessage message = inbox.GetMessage(item);
-                    emailProcessor(ConvertMessageToReadEmail(message, filters));
+                    if (filters.HasAttachments == EmailReadFilters.eHasAttachmentsFilter.Either)
```

Comment:
All Conditions are executing same line of code after once true. Please try merging all conditions together to avoid duplication of code